charms/istio-pilot/tests/unit/test_istioctl.py

```diff
@@ -64,11 +70,32 @@ def test_istioctl_install(mocked_check_call):
         ISTIOCTL_BINARY,
         "install",
         "-y",
-        "-s",
-        f"profile={PROFILE}",
-        "-s",
-        f"values.global.istioNamespace={NAMESPACE}",
     ]
+    expected_call_args.extend(EXPECTED_ISTIOCTL_FLAGS)
+
+    mocked_check_call.assert_called_once_with(expected_call_args)
+
+
+def test_istioctl_install_extra_flags(mocked_check_call):
```

Comment:
Should we add a similar test for upgrade as well?  Or is that overkill?

I know upgrade gets its istio install flags from the same property as install does, so atm it feels well tested.  But that feels like we get our test coverage by knowing an implementation detail that isn't guaranteed.  wdyt?  If it feels like overkill, I'm fine ignoring

Reply:
Yeah, I had the same feeling as you, but decided not to add one exactly because of the things you are mentioning. In the end we will be testing that the expected args are passed, which is already tested in this test case.

Reply:
Yeah fine with me!